src/contract.ts

```diff
@@ -43,14 +42,24 @@ export class CWContractInstance {
   public vm: VMInstance;
   public executionHistory: any[] = [];
 
+  private _bech32Prefix: string;
+  private _height: number;
+  private _time: number;
+  private _chainId: string;
+
   constructor(
-    private _chain: CWChain,
+    chain: CWChain,
     public contractAddress: string,
     public contractCode: CWContractCode,
     public storage: IIterStorage = new BasicKVIterStorage()
   ) {
+    this._bech32Prefix = chain.bech32Prefix;
+    this._height = chain.height;
```

Comment:
this most likely won't work. it locks this particular contract instance to a specific point in time in the blockchain & will cause (logic) errors as we instantiate or execute. perhaps we should pass an `Env` to `instantiate` & `execute` instead of building it internally instead?